analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/LocksReleasedAllPathsBase.cs

```diff
@@ -113,8 +115,12 @@ public override void ExecutionCompleted()
             }
         }
 
-        protected static bool ShouldExecuteFor(SyntaxToken identifier) =>
-            identifier.Text.Contains("Exit") || identifier.Text.Contains("ReleaseMutex") || identifier.Text.Contains("ReleaseReaderLock");
+        public override bool ShouldExecute()
+        {
+            var collector = new LockAcquireReleaseCollector();
+            Visit(NodeContext.Node, collector);
```

Comment:
There's a bug in the way it is used. If we call it with lambda entry point, the first thing the walker does is that it stops walking the tree. We can avoid it by invoking like this
```
            foreach (var child in NodeContext.Node.ChildNodes())
            {
                Visit(NodeContext.Node, collector);
            }
```
but then it doesn't make sense to recreate the walker instance inside :/